src/main/js/controls/Bubble/helpers/colorGradient.js

```diff
@@ -44,3 +44,52 @@ export const generateColor = (dataTarget) => {
 
     return huePaletteList;
 };
+
+/**
+ * Generates color range for the given color for a single dataset
+ *
+ * @private
+ * @param {object} dataTarget - data for the bubble graph
+ * @returns {d3.scale} - returns function to get color for specific bubble
+ */
+export const generateColorSingleDataset = (dataTarget) => {
+    let radiusData;
+
+    if (utils.isUndefined(dataTarget.weight)) {
+        radiusData = dataTarget.values.map((element) => element.y);
+    } else {
+        radiusData = dataTarget.values.map((element) =>
+            bubbleScale(dataTarget)(element.weight)
+        );
+    }
+
+    let color = [];
```

Comment:
Can you please initialize everything at one place i mean at the start of method?

Reply:
Fixed here: 
https://github.com/cerner/carbon-graphs/pull/308/commits/c8a7c1526c25831c4090e4e4dcef279d75cc8fb7